plugin.go

```diff
@@ -0,0 +1,283 @@
+/*
+Drone plugin to upload one or more packages to Bintray.
+See README.md for usage.
+Author: Archit Sharma December 2019 (Github arcolife)
+Previous: David Tootill November 2015 (GitHub tooda02)
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/arcolife/jfrog-client-go/bintray"
+	"github.com/arcolife/jfrog-client-go/bintray/auth"
+	"github.com/arcolife/jfrog-client-go/bintray/services"
+	"github.com/arcolife/jfrog-client-go/bintray/services/packages"
+	"github.com/arcolife/jfrog-client-go/bintray/services/repositories"
+	"github.com/arcolife/jfrog-client-go/bintray/services/versions"
+	"github.com/pkg/errors"
+	// "github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+)
+
+type (
+	// Package - struct fields must be public in order for unmarshal to
+	// correctly populate the data.
+	Package struct {
+		Package string                `yaml:"package"`
+		Config  packages.Params       `yaml:"config"`
+		Upload  services.UploadParams `yaml:"upload"`
+	}
+
+	// Repo - struct fields must be public in order for unmarshal to
+	// correctly populate the data.
+	Repo struct {
+		Name     string              `yaml:"name"`
+		Subject  string              `yaml:"subject"`
+		Config   repositories.Config `yaml:"config"`
+		Packages []Package           `yaml:"packages"`
+	}
+
+	// Bintray - struct fields must be public in order for unmarshal to
+	// correctly populate the data.
+	Bintray struct {
+		Repos []Repo `yaml:"repos"`
+	}
+
+	// BintrayConfig -> entrypoint for all configs
+	BintrayConfig struct {
+		Threads       int  `yaml:"threads"`
+		Cleanup       bool `yaml:"cleanup"`
+		Bintray       `yaml:"bintray"`
+		Username      string `yaml:"username"`
+		APIKey        string `yaml:"api_key"`
+		GPGPassphrase string `yaml:"gpg_passphrase"`
+	}
+
+	Plugin struct {
+		BintrayConfig      `yaml:"bintray_config"`
+		ServicesManager    *bintray.ServicesManager
+		BintrayConfigPath  string   `yaml:"bintray_cfg_path"`
+		Version            string   `yaml:"verison"`
+		UploadPackage      bool     `yaml:"uploadpackage"`
+		SignPackageVersion bool     `yaml:"signpackageversion"`
+		PublishPackage     bool     `yaml:"publishpackage"`
+		CalcMetadata       bool     `yaml:"calcmetadata"`
+		ShowPackage        bool     `yaml:"showpackage"`
+		Envs               []string `yaml:"envs"`
+	}
+)
+
+const defaultLicense = "Apache 2.0"
+
+func (p Plugin) Exec() error {
+	// var (
+	// 	files []string
+	// )
+	// fmt.Printf("Drone Bintray Plugin built from %s\n", p.Version)
+	cSig := make(chan os.Signal)
```

Comment:
We aren't doing that for any plugin and I can't clearly see why we should introduce that here.